### DISCUSSION TITLE
How to get back the ID of the customer when using checkout

### DISCUSSION THREAD

**Discussion by kieron** [jofftiquez/vue-stripe-checkout#180] [Q&A]
Hi there, I plan on using the customer portal for when a customer actually becomes a customer, and naturally, to use this you need your customer ID, and naturally, for this you need to be a customer > so I figured id go with the stripe-checkout component, but unless I am mistaken, I can't see any return when it directs you to your success page?

Perhaps I need to use a different method, I am just not sure which method I need? I'm aware I'm not doing anything magical or unique, so I must be overthinking it. :)

I am just trying to get an MVP going with 3 tiered pricing, but yeah, in order to allow management of a customer I need that customer ID to at some point hit either my own API, or my own front end - so what do people generally do in this case?

Thanks!

**Comment by jofftiquez** [comment]
Hi @kieron first of all I would like to apologize because I didn't see this post. Thank you for using the discussions section.

I have a question regarding this.

1. What do you mean with "I can't see any return when it directs you to your success page?"

If I get you correctly you like to create a customer instance. You don't need to use the `stripe-checkout` component for this. All need to do is:

1. Create a form in YOUR FRONTEND to get user's billing info, (credit card is not required)
2. Send the user info to YOUR BACKEND
3. YOUR BACKEND will do a request to STRIPE API to create a Customer, you can utilize the `Customer#metadata` to create a reference to the customer. Using this reference you can query the Customer data from Stripe anytime you want. You can use email or the uid of the customer from YOUR DATABASE.
4. Now you have a customer

When it's time to ask the user to subscribe or upgrade, you can create a Source using VueStripeCard component to collect their tokenized payment information such i.e. credit card. Send that Source to YOUR BACKEND and connect it to the Customer in stripe, this way, when you use that customer to bill a subscription, you will be able to collect the payment. 

There's more way to get around with this, but this is what I currently have in mind.

**Reply by kieron** [reply]
Hey @jofftiquez no problem at all for the delay :)

I do have it working now.

You were right though, I didn't need to use the `stripe-checkout` component at this point!

Thanks! 💯

**Reply by jofftiquez** [reply]
@kieron glad you worked it out. Let me know if need any help in the future. Cheers!
